lib/helpers.js

```diff
@@ -127,11 +130,58 @@ const parseResults = (editor, results, filePath, showIgnored) => {
   return toReturn;
 };
 
+function isDirectory(dirPath) {
+  try {
+    return fs.statSync(dirPath).isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+export function findStylelintDirectory(modulesDir) {
+  const stylelintDir = path.join(modulesDir || '', 'stylelint');
+  if (isDirectory(stylelintDir)) {
+    return stylelintDir;
+  }
+  return stylelintLocalPath;
+}
+
+export function getStylelintFromDirectory(modulesDir) {
+  const stylelintDir = findStylelintDirectory(modulesDir);
+  try {
+    // eslint-disable-next-line import/no-dynamic-require
+    return require(stylelintDir);
```

Comment:
`stylelint` currently takes a [long time to require](https://github.com/stylelint/stylelint/issues/2454), this should be cached to only incur that penalty once instead of on each lint request.